tests/test_constants.py

```diff
@@ -99,5 +99,19 @@ def test_request_type(self):
             self.fail("Couldn't instantiate enum")
 
 
+class TestDataSource(TestSentinelHub):
+    def test_adding_custom_datasource(self):
+        collectionid_datasourcename_wfsid = (
+            ('0000d273-7e89-4f00-971e-9025f89a0000', 'BYOC_0000d273-7e89-4f00-971e-9025f89a0000',
+             'DSS10-0000d273-7e89-4f00-971e-9025f89a0000'),
+        )
+        for collectionId, dataSourceName, wfsId in collectionid_datasourcename_wfsid:
+            datasourcename_tested = DataSource(collectionId).name
+            wfsid_tested = DataSource.get_wfs_typename(DataSource(collectionId))
+            self.assertEqual(dataSourceName, datasourcename_tested, msg="Expected {}, got {}".
+                             format(dataSourceName, datasourcename_tested))
+            self.assertEqual(wfsId, wfsid_tested, msg="Expected {}, got {}".format(wfsId, wfsid_tested))
+
```

Comment:
Here you could also test calling various methods of `DataSource` class for BYOC data source constant (e.g. `is_sentinel1`, `is_timeless`, ...).